### DISCUSSION TITLE
Develop automated methods for documentation QC and standardization

### DISCUSSION THREAD

**Discussion by mattroumaya** [Gilead-Public/gsm#331] [Ideas]
Since a lot of the `*_Map` and `*_Assess` functions are very similar, and contain identical parameters, I was thinking that there are probably standard checks that we'll want to do for each new release, or at least for major releases. 

I think that as long as the way to run and use the functions is documented, it'll save dev and QC time in the future by running some standard checks and then addressing the issues, without having to compare across .R files or coming up with new ways to compare. 

Some ideas are:

1. Produce [params.xlsx](https://github.com/Gilead-BioStats/gsm/files/8363900/params.xlsx) to make sure that params are all documented in the same way across functions:
```r
library(rvest)
library(tidyverse)
library(openxlsx)

htmlTable <- function(x) {
  rvest::read_html(x) %>% 
    rvest::html_nodes("table") %>% 
    rvest::html_table() %>% 
    bind_rows() %>% 
    mutate(Function = X1[1]) %>% 
    slice(-1) %>% 
    select(Function, Param = X1, Description = X2)
}

docs_list <- map(list.files("./man/"), ~ tools::Rd2HTML(paste0("./man/",.), out = tempfile("tmp")))

params <- map_df(docs_list, ~htmlTable(.))

# openxlsx::write.xlsx(params, "params.xlsx")
```
2. Run `spellcheck()` and produce a table, and file an issue to address, using: 
```r
ignore <- ignore <- c("ADAE", "adam", "ADaM", "adsl", "ADSL", "AE", "AEs", "Bugfix", 
            "Chisq", "clindata", "CMD", "Codecov", "CONSCAT", "CONSDAT", 
            "CONSYN", "CRF", "customizations", "dataChecks", "DCREASCD", 
            "dev", "devtools", "df", "dfADAE", "dfADSL", "dfAE", "dfAllChecks", 
            "dfConsent", "dfCONSENT", "dfDemog", "dfDisp", "dfDomain", "dfFindings", 
            "dfFlagged", "dfIE", "dfInput", "dfPD", "dfRDSL", "dfSUBJ", "dfSubject", 
            "dfSummary", "dfTransformed", "directionality", "Disp", "EMA", 
            "EventCount", "FilterDomain", "GCP", "https", "IDCol", "IDEs", 
            "ie", "IECAT", "IEORRES", "KRI", "KRIs", "lParams", "lTags", 
            "mainconsent", "md", "mergeSubjects", "na", "NaN", "nThreshold", 
            "params", "PDs", "pkgdown", "poisson", "pre", "PredictedCount", 
            "PRs", "Pvalue", "PValue", "RandDate", "RBM", "RDSL", "readme", 
            "reprex", "roxygen", "Roxygen", "safetyData", "SDTM", "SiteID", 
            "SITEID", "str", "strCategoryCol", "strCol", "strColumn", "strCountCol", 
            "strDateCol", "strEndCol", "strExposureCol", "strFunctionName", 
            "strIDCol", "strRandDateCol", "strRateCol", "strReason", "strResultCol", 
            "strSiteCol", "strStartCol", "strTimeOnStudyCol", "strTimeOnTreatmentCol", 
            "strTreatmentEmergentCol", "strTypeCol", "strValueCol", "strValueColumn", 
            "styler", "SubjectID", "SUBJID", "testthat", "themockup", "ThresholdCol", 
            "ThresholdHigh", "ThresholdLow", "tidyverse", "TimeOnStudy", 
            "TimeOnTreatment", "TotalCount", "TotalExposure", "TRTEDT", "TRTSDT", 
            "USUBJID", "valtools", "vNACols", "vRequired", "vThreshold", 
            "vUniqueCols", "wilcoxon", "yaml", "YAML")

devtools::spell_check() %>% 
  tidyr::unnest(cols = found) %>% 
  dplyr::filter(!word %in% ignore) %>% 
  gt::gt() %>% 
  gt::as_raw_html()
```

**Comment by mattroumaya** [comment]
Tagging @jwildfire for review whenever, meant to bring up in scrum but we had a lot to discuss today. This is low priority but might be useful for future releases.

**Comment by jwildfire** [comment]
I like this idea. The params table especially. Might be nice to have the table generated automatically via a github action whenever we have a PR to `main` and then have review of the table as part of the checklist in the PR template.

**Reply by mattroumaya** [reply]
If we used something like [this](https://github.com/r-lib/actions/tree/master/examples#render-rmarkdown), we could make a HTML report to aid in QC. I tried to attach here but GH won't let me - I'll email a draft to you to see if it's something you think would be useful. 

[Here](https://github.com/Gilead-BioStats/gsm/blob/qc-report/inst/qc/qc.rmd)'s the branch with the `.rmd` if you're interested in checking that out.
